### PR TITLE
Defines Categories endpoint

### PR DESCRIPTION
**Pull request by docStonehenge**
WIP

### CODE REVIEW

File: lib/myfinance.rb [hunk 1]
```diff
@@ -16,13 +16,15 @@
...
 require "myfinance/entities/attachment"
+require "myfinance/entities/category"
 
 require "myfinance/resources/base"
 require "myfinance/resources/entity"
 require "myfinance/resources/financial_account"
 require "myfinance/resources/payable_account"
 require "myfinance/resources/receivable_account"
 require "myfinance/resources/attachment"
+require 'myfinance/resources/category'
```
[the remark this answers is not in the document]
Comment (leandrost) [reply lib/myfinance.rb]: I agree with hound on this one

File: spec/lib/myfinance/client_spec.rb [hunk 2]
```diff
@@ -60,4 +60,11 @@
       subject.attachments
     end
   end
+
+  describe '#categories' do
```
[the remark this answers is not in the document]
Comment (leandrost) [reply spec/lib/myfinance/client_spec.rb]: +1

File: spec/lib/myfinance/entities/category_collection_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
```
[the remark this answers is not in the document]
Comment (leandrost) [reply spec/lib/myfinance/entities/category_collection_spec.rb]: +1 for double quotes

File: spec/lib/myfinance/client_spec.rb [hunk 4]
```diff
@@ -60,4 +60,11 @@
...
+    it 'instantiates a new Myfinance::Resources::Category' do
```
[the remark this answers is not in the document]
Comment (leandrost) [reply spec/lib/myfinance/client_spec.rb]: +1

File: spec/lib/myfinance/resources/category_spec.rb [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  describe "#destroy" do
+    context "with success" do
+      it "destroy a category successfully" do
+        expect(client.categories.destroy(532314)).to be_truthy
+      end
+    end
+
+    context "with error" do
+      it "raises Myfinance::RequestError with 404 status code" do
+        expect {
```
[the remark this answers is not in the document]
Comment (leandrost) [reply spec/lib/myfinance/resources/category_spec.rb:110]: +1

**Review by leandrost — CHANGES_REQUESTED** [review]
Looks ok for me, but would be good take a look at these comments made by Hound.

**Review by vagnerzampieri — CHANGES_REQUESTED** [review]
Hound and conflicts.

**Review by lmansur — APPROVED** [review]

**Comment by rodrigotassinari** [comment]
:shipit: released on [version 0.5.0](https://rubygems.org/gems/myfinance/versions/0.5.0)

**Pull request merged**